### DISCUSSION TITLE
نحوه اجرا پروژه

### DISCUSSION THREAD

**Discussion by parsa-programmer** [mohammad-ashrafi2020/DigiLearn#4] [Q&A] [answered]
چطوری باید پروژه رو اجرا کنم ؟!

**Comment by mohammad-ashrafi2020** [comment, marked as answer]
اول از هر چیز باید نسخه 6 و 7 و 8 دات نت روی سیستم اتون نصب باشه که میتونید یا از سایت ماکروسافت دانلود اش کنید یا با Vs Installer نصب اش کنید
نرم افزار هایی که لازم داریم :
Visual Studio
Docker ( اختیاری)
Sql Server
Redis
MongoDb
Rabbit Mq

پروژه رو که باز کنید باید وارد appsettings.json بشید و آدرس sql Server رو تغییر بدیت به کانکشن دیتابیس خودتون و اگر redis روی port پیشفرض خوداش نصب نیست باید توی program.cs آدرس اش رو تغییر بدین و از پروژه اجرا بگیرید و تمام
البته برای اینکه دیتابیس هاتون ساخته بشه لازمه که این دستور رو اجرا کنید :
Update-Database -context اسم کانتکس
ما توی پروژه چندین دیتابیس داریم پس برای هر ماژول که از دیتابیس sql server ستفاده میکنه یه Context جدا ساختیم و شما برای اینکه همه دیتابیس ها به درستی ساخته بشن باید دستور بالا رو برای اسم تمام کانتکس ها صدا بزنید ( برای هر کانتکس جدا صدا زده بشه )

البته همه این موارد رو توی دوره توضیح دادم و قدم به قدم کار رو پیش بردیم
